sourcemap/mapper.go

```diff
@@ -63,5 +64,10 @@ func (m *SmapMapper) Apply(id Id, lineno, colno int) (*Mapping, error) {
 }
 
 func (m *SmapMapper) NewSourcemapAdded(id Id) {
+	_, err := m.Accessor.Fetch(id)
+	if err == nil {
+		logp.NewLogger("sourcemap").Warnf("Overriding sourcemap for service %s version %s and file %s",
+			id.ServiceName, id.ServiceVersion, id.Path)
+	}
```

Comment:
I think this should be done after `m.Accessor.Remove(id)` as we use negative caching. The cache should first be emptied for this entry, otherwise it could happen that a `nil` value is read from cache (no error) although no entry is there.

Reply:
ah, missed that `Fetch` looks up in the cache first. Good catch, fixed now

Reply:
hmm, i don't think that works... If I `remove` first and `fetch` later, the `fetch` will put a `<id, nil>` entry in the cache, undoing the previous `remove` and leaving the cache inconsistent. So `remove` necessarily needs to happen later.
I changed it again now, but it is a bit fiddly. does this makes sense to you?